src/init.lua

```diff
@@ -1,14 +1,47 @@
 local Internal = require(script.Internal)
+local PromiseTypes = require(script.PromiseTypes)
 
 local internal = Internal.new(true)
 
+export type DataStoreService = {
+	GetDataStore: (name: string) -> GlobalDataStore,
+	GetRequestBudgetForRequestType: (requestType: Enum.DataStoreRequestType) -> number,
+}
+
+export type PartialLapisConfig = {
+	saveAttempts: number?,
+	loadAttempts: number?,
+	loadRetryDelay: number?,
+	showRetryWarnings: boolean?,
+	dataStoreService: DataStoreService?,
+	[any]: nil,
+}
+
+export type CollectionOptions<T> = {
+	defaultData: T,
+	migrations: { (any) -> any }?,
+	validate: (T) -> (boolean, string?),
+	[any]: nil,
+}
+
+export type Collection<T> = {
+	load: (self: Collection<T>, key: string) -> PromiseTypes.TypedPromise<Document<T>>,
+}
+
+export type Document<T> = {
+	read: (self: Document<T>) -> T,
+	write: (self: Document<T>, T) -> (),
+	save: (self: Document<T>) -> PromiseTypes.TypedPromise<()>,
```

Comment:
These promises don't return a value, so it'd probably be better to just have it return `PromiseTypes.Promise`

Reply:
I think it's still useful to use `TypedPromise` here so that if you chain with a `andThen` for example it knows the argument passed is `()` and not `any`.